### ISSUE TITLE
Killed by Blob Lieutenant Shadow without warning and without recourse

### ISSUE THREAD

**Issue by Inglonias** [CleverRaven/Cataclysm-DDA#70951]
### Is your feature request related to a problem? Please describe.

I was having a really good game today - I was a little more than a week in, and I had just bought welding equipment from the refugee center. I was walking home when suddenly I got a message about the night closing in, and then I was swarmed by something and died.

Further research reveals the culprit was #66869 . Apparently the lieutenant is weak to light? I don't feel like I had a way to know that.

In addition, I was in the middle of a field and had no cover. I basically had no recourse here.

### Solution you would like.

While I really like this idea, it really did feel like the game just said "fuck you" and killed me without any sort of warning. I thought we were trying to move away from that?

I did hear some mysterious shrieks when looting a gun store - That might have also been this. I honestly don't know.

I would appreciate fewer enemies in this encounter, something that I can do to stop it or survive it, or some other sort of recourse. This feels unacceptable to me, and I feel inclined to simply mod it out. Games that go this well are rare enough as it is for me.

### Additional context

![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/2125926/39001a57-a9ae-4eac-a6d9-25e3e34e19f6)

**Comment by Inglonias** [comment]
If there was something I could have done here to escape or a limited amount of time in the encounter that I was otherwise unaware of, please let me know.

**Comment by Inglonias** [comment]
The main issue I faced was that I had ZERO context for what was happening. This is actually quite manageable once you have the necessary knowledge - Notably, once you know that the lieutenant and his minions are weak to light.

After some experimentation, I have a few suggestions:

1. It is very unclear that the amalgamations will continue to spawn until you escape to a safe place, or even what a safe place is - I don't know how you would communicate this to the player.
2. There is very little suggesting that light is so effective against this thing - Recommend more strongly emphasizing text that tells you how dark it is, or how any light that's there is gone, or something. Just start referencing the concept of light. A lot.
3. Amalgamation spawns are too frequent and too numerous. Cutting the rate and/or number of spawns per spell cast would help reduce the sheer panic of the situation.
4. Suggest we more clearly emphasize initial spawning. Make a text box analogous to a portal storm.
5. There does not appear to be any way to harm the target. At all. I feel that it should be vulnerable to high energy situations - fire, energy weapons, heat, light.

**Comment by Inglonias** [comment]
Pinging @RenechCDDA with the above suggestions. This is a really cool boss fight. I like it a lot, it's just a bit too dangerous.

With your permission, I would like to take a crack at rebalancing it a little bit.

**Comment by carlarctg** [comment]
I agree wholeheartedly. When I had this encounter for the first time, I had no idea what to do and tried to both fight it and run away. Fighting it did nothing because it annoyingly runs away from you and takes no damage from the vast majority of things.
Then when I tried to run away I had no stamina and lost, not that it would've mattered since the swarming amalgamations are faster than you.

If I'm being beset by a caster enemy then I'm obviously going to run it down to stop it from summoning more, like the zombie master or necromancer.

All of this was, if I recall correctly, with a flashlight enabled. It just didn't do enough to help me.

I had basically the same experience as Inglonias, with the only difference being that I was so annoyed that I refused to accept that death and cheated to keep my run going.

This is, fundamentally, not a good execution for the concept, which is dubious enough as is. Either you know what it does and are fine or you don't and you die without any possible hint or knowledge on how to survive.

I recall a comment, I believe on the CDDA reddit on this topic, that talked about a fanmade version of a Civlization game in which the 'Easy' AI was punishingly hard, because the author was an experienced player and could play around it quite easily, while a player inexperienced with Civilization would be very quickly overwhelmed by the enemy. This is basically the same scenario.

To add insult to injury, there's no reward for the encounter, no incentive, no cool materials or loot or rewards like portal storms or some tough zombies give. It basically feels like an encounter designed specifically to ruin your game. There's nothing to it but running away, and while this could be well-executed, it very much is not at the moment.

**Comment by Inglonias** [comment]
I feel that the lack of a reward is not a probkem. Defeating a threat this big should be its own reward - being able to travel at night safely.

**Comment by fairyarmadillo** [comment]
I mentioned this in the thread but it bears repeating here - you got spammed with a ton of messages that you ignored and you didn't carry a flashlight at night in a horror game set in an apocalypse. A flashlight totally negates the danger. This is a situation where it's pretty reasonable that you would just die.

**Comment by RenechCDDA** [comment]
> Pinging @RenechCDDA with the above suggestions. This is a really cool boss fight. I like it a lot, it's just a bit too dangerous.
> 
> With your permission, I would like to take a crack at rebalancing it a little bit.

One thing that I want it to have but haven't gotten around to is the warning.

Basically, `EOC_LIEUTENANT_SPAWN_SHADOW` would be given one additional condition (`Shadow_warnings >= 3` or something), and an additional false_effect to fire another eoc - `EOC_SHADOW_SPAWN_FAILED_GIVE_WARNING` or something less verbose. EOC_SHADOW_SPAWN_FAILED_GIVE_WARNING would have the exact same conditions, except the RNG 1-in-6 and the warnings check. Meaning it would only actually fire at night, in forests and fields, and when shadow *could* spawn. But didn't, either because you haven't been warned enough, or because you got lucky in the 1-in-6 chance.

The primary purpose of the warning EOC would be to display snippets, and of course those need to be written. The snippets can serve as part of the guard rails for the encounter, telling the player it's not safe to be out at night and that they would be much more comfortable(safe) indoors. There's also been some repeated suggestions and ideas about some sort of mock first encounter which is much less deadly, or otherwise showing off a little bit of the way it works without hitting you with the full 'boss'. Such as spawning a (single) group of amalgamations with slightly longer timespans and the usual terrain destruction, along with a snippet.

There are problems with this approach. The spawn length check is long enough that a player could hang out next to safety, get the warning, and then have ~4 hours to go ham in pure safety. Ideally the timer range would be pushed out in both directions so it's something like `30 minutes, 9 hours` (giving roughly the same average, but much more variance - push your luck at your peril!).

(Note that my proposed setup means that warnings may continue to accumulate even past the first 3, as long as shadow is alive and misses the spawn chance. So if a mock spawn were introduced as part of the warnings this needs to be accounted for. Variables flying everywhere, ain't it great?)

Just to cap this off: You don't *need* my blessing to do anything. But I would be happy to give my opinion if you want to try adding to the EOCs.

**Comment by RenechCDDA** [comment]
And just to write this down, since I have no further excuse for not having done this yet: I would like to reduce shadow's speed to 190 after giving it a teleport-away-on-being-hit EOC. As you've mentioned, it can be very difficult to actually harm it. Hitting it with a flashlight gives photophobia, bringing it down to 100 speed... but that's *your* speed too, so to catch it in melee you either need to be faster than 100 speed (typically not a thing), let it run over some bad terrain(very situational, very iffy), or otherwise completely break the encounter.

Having the teleport-away-on-being-hit EOC means that the fight is much more structured. You should have a definite speed advantage inside bright light range, and that means you can close and hit it - but it will poof away, and then you start again from the top but with it having less HP. There's defined phases, and the fight is much more reasonable in case you're trying to kill it. And if you're trying to run away from it, it having even less speed in light is never going to hurt.

**Comment by fairyarmadillo** [comment]
> > There does not appear to be any way to harm the target. At all. I feel that it should be vulnerable to high energy situations - fire, energy weapons, heat, light.
> 
> It has no heat armor, so the vanilla (or AFS) laser weapons should obliterate the thing in short order. It's purposefully intelligent though, so it will avoid like... a fire on the ground. A directed flamethrower is probably a reasonable way to kill it though? I would need to check some numbers to see how much damage the flamethrower/other sources of fire damage actually does. It mostly has 25 fire armor because of _magiclysm etc_

I'm not sure what you mean here, there's no difference between fire and heat armor in this game. They're two different words used interchangeably for the same kind of damage. There is a second kind of damage that only NPCs and the player get from the hot air field, but that's unrelated and no armor stat protects from that.

Regardless, flamethrowers are pretty rare and not something anyone would have at hand in any given scenario, and lasers are a rare endgame item that no character should ever be expected to have. I don't think it's at all appropriate to expect the player to have one for an encounter that's forced on them a week into the game. If the enemy is going to ambush them, they should be able to fight it with a variety of weapons because we don't want to force a meta where you have to have this one specific kind of weapon at all times or it's game over.

Requiring heat damage is something that would be much more appropriate for a location which the player is meant to assault, such as an enemy base or a boss in a known location. That way planning for the encounter and hunting down the required items becomes part of a quest chain, not a punishment for not playing in one specific way.

**Comment by fairyarmadillo** [comment]
Furthermore if Magiclysm is causing you to need to adjust the balance of the monster, then its jsons should be updated in Magiclysm. Balancing factors like that shouldn't be left in the core game.

**Comment by RenechCDDA** [comment]
> I'm not sure what you mean here, there's no difference between fire and heat armor in this game. They're two different words used interchangeably for the same kind of damage

I found the issue. HHG enumerates the heat armor as `fire`(armor). I trusted it more than I trusted my memory.

That's an issue on their side, I'll see about resolving it separately.

**Comment by RenechCDDA** [comment]
> Regardless, flamethrowers are pretty rare and not something anyone would have at hand in any given scenario, and lasers are a rare endgame item that no character should ever be expected to have. I don't think it's at all appropriate to expect the player to have one for an encounter that's forced on them a week into the game. If the enemy is going to ambush them, they should be able to fight it with a variety of weapons because we don't want to force a meta where you have to have this one specific kind of weapon at all times or it's game over.

Yes, which is why I am speaking of *giving more possibilities*, not pigeonholing players into new ones. With the speed adjustment, melee is a much more viable option. With the (however it is resolved) fire discrepancy, a flamethrower is an option. High explosives, 50cal turrets, and the like also remain an option. (This is one of the few cases the big guns get to be useful, so I definitely want to keep that open!)

**Comment by fairyarmadillo** [comment]
> > I'm not sure what you mean here, there's no difference between fire and heat armor in this game. They're two different words used interchangeably for the same kind of damage
> 
> I found the issue. HHG enumerates the heat armor as `fire`(armor). I trusted it more than I trusted my memory.
> 
> That's an issue on their side, I'll see about resolving it separately.

I should mention that the game internally also uses the terms interchangeably for the same type of damage, and when it does it's always referring to laser beams and flames, not high-temperature air. A _lot_ of people get confused by this, and if it even fooled you it may be worth looking into changing how it's named in the code.

There's a similar thing with stab/pierce damage, which is the same thing but may confuse people because there's also armor-piercing, which is different.

**Comment by Inglonias** [comment]
> I mentioned this in the thread but it bears repeating here - you got spammed with a ton of warning messages over the course of hours that you ignored, and you didn't carry a flashlight at night in a horror game set in an apocalypse. A flashlight totally negates the danger. This is a situation where it's pretty reasonable that you would just die.

A warning message that you can't understand is useless and I agree in hindsight that it's reasonable that I would die in this situation.

The key rebalance I'm thinking of here is to make the warning messages more understandable and more likely to stick out to the average player. I did have a head lamp but at no point did it occur to me to use it - ordinarily lighting yourself up at night is a bad idea, not a good one.

**Comment by fairyarmadillo** [comment]
> A warning message that you can't understand is useless and I agree in hindsight that it's reasonable that I would die in this situation.
> 
> As for why those messages were ignored, I was auto traveling to and from a place I thought to be safe. I wasn't on my guard and quite literally on autopilot. If I really thought this death was entirely unjustified, I would have reverted it. I didn't.
> 

I don't agree that the warning messages aren't understandable. They're very strange and put you on alert that something might be up. We don't need to spell every detail out to the player before it happens, at that point we may as well just have an NPC come out of the woods and kill the monster for them. Players should be on the lookout for the unexpected.

The player should absolutely be knocked out of autotravel by the warning messages. I personally never use autotravel because there are a lot of things that should stop you but don't, and you can easily get killed without understanding why.

**Comment by Cenness** [comment]
On tangential matters - blob shadow should probably be disabled or at least postponed till midgame (a month or two instead of a week) for 0.H stable

**Comment by RenechCDDA** [comment]
> On tangential matters - blob shadow should probably be disabled or at least postponed till midgame (a month or two instead of a week) for 0.H stable

Pretty strong disagree here, although again: opinion. The week it gives you is just to avoid ganking you if your start isn't in a place where you can shelter, past that it is expected you not stick your nose in places where you might get bit. After the week, it is very much an "opt-in" sort of encounter.

**Comment by Inglonias** [comment]
> Pretty strong disagree here, although again: opinion. The week it gives you is just to avoid ganking you if your start isn't in a place where you can shelter, past that it is expected you not stick your nose in places where you might get bit. After the week, it is very much an "opt-in" sort of encounter.

I agree that this is definitely what you were going for, but if you don't know what it is or that it exists, it very much is not opt-in. That's the main problem I want to solve with #70971, along with giving players an avenue to kill the thing with fire, because difficulty-wise, it's actually not that bad.

**Comment by Cenness** [comment]
> at least postponed


> Pretty strong disagree here

What I was meant to say is that Rubik kinda mention that world changes as times passes and exodii don't risk wobbling into old cataclysms. I thought this lieutenant was an element of that, lore wise. And if it is, shadow should not spawn in the very first month. Am I wrong in that assumption?

**Comment by RenechCDDA** [comment]
> I thought this lieutenant was an element of that, lore wise. And if it is, shadow should not spawn in the very first month. Am I wrong in that assumption?

Lore-wise the (rather erroneously named) lieutenants are basically just agents of destruction. There is not some sort of actual military hierarchy, they are not necessarily subordinate to the 'blob', and the lieutenants might hate the blob just as much as you do. One of the rather popular ideas that has floated around is that **the player** could be a blob lieutenant. A notable number of players play psychopathic characters, butchering and eating other humans. They also do things like break into secure facilities, or "raid the hub" (literally murdering one of the few bastions of civilization left) for their trinkets.

Now imagine if the player was dropped into a world as part of the initial invasion.

The presence of the blob lieutenants is specifically pre-cataclysm, or what we refer to as cataclysm day (the day you start playing). They're there to screw up the place! And they're good at it.

**Comment by Voideka** [comment]
> > A warning message that you can't understand is useless and I agree in hindsight that it's reasonable that I would die in this situation.
> > As for why those messages were ignored, I was auto traveling to and from a place I thought to be safe. I wasn't on my guard and quite literally on autopilot. If I really thought this death was entirely unjustified, I would have reverted it. I didn't.
> 
> I don't agree that the warning messages aren't understandable. They're very strange and put you on alert that something might be up, giving you as much information as your character might have, which is in this case enough to know that tonight it will not be business as usual. We don't need to spell every detail out to the player before it happens. Players, like their characters, should be on the lookout for the unexpected.
>

I think adding this additional warning is great, but I do still think the spawn messages are misleading. It requires the player to feel that EoC popup messages are always negative things to be afraid of, which I am not a fan of personally.

I also was really hoping the various other issues I brought up in #69673 would get some attention, such as the event/monsters spawning literally indoors.

**Comment by RenechCDDA** [comment]
> I also was really hoping the various other issues I brought up in #69673 would get some attention, such as the event/monsters spawning literally indoors.

It is not reasonably possible to stop them(amalgamations) from spawning indoors at this time. When it tries to spawn them it searches a small radius around the player for a grass tile and spawns monsters in an area around the found grass tile. The spawning is entirely normal functions, and doesn't "flood fill" or check reachability of the resulting tiles it places monsters on. (Thus, this occurs when the selected grass tile is right next to a single-tile wall)

Now I said "reasonably possible", and the reason is that it could be reconfigured to instead spawn singular monsters, but repeat the loop several times (to spawn multiple single monsters on valid terrain), and then only after to apply the terrain transformation.

But that's probably hundreds of lines of relatively fragile EOC code that would be a nightmare to maintain/debug. If you know of a good way to do this, I'd be happy to hear it. But it's just not that simple.

**Comment by Voideka** [comment]
I only did about 20 tests, but adding "outdoor_only": true to spawn section of the life drain EoC seems to prevent enemies from being spawned indoors.

**Comment by RenechCDDA** [comment]
> "outdoor_only"

I might be an idiot.

I will try to test it this evening and get it into #70990 (assuming it works as desired)

**Comment by Inglonias** [comment]
> > "outdoor_only"
> 
> I might be an idiot.
> 
> I will try to test it this evening and get it into #70990 (assuming it works as desired)

Don't feel too bad. This game's code is massive and there's a lot that doesn't quite make sense. I could regale you with tales of the massive, game-breaking difference between `spawn_monster` and `spawn_monsters`

**Comment by Inglonias** [comment]
I am looking into adding basic enchantment support for monsters. I hope to have something soon!

**Comment by Inglonias** [comment]
With the additions of better warnings as well as weakening the monster through enchantments as intended by renech, this issue can now be considered closed.
